api/s2n.h

```diff
@@ -241,12 +241,35 @@ S2N_API extern int s2n_init(void);
 S2N_API extern int s2n_cleanup(void);
 
 /**
- * Create a new s2n_config object. This object can (and should) be associated with many connection objects.
+ * Creates a new s2n_config object and initializes it with default system certificates. This object
+ * can (and should) be associated with many connection objects.
  *
- * @returns returns a new configuration object suitable for associating certs and keys. 
+ * Use `s2n_config_new_minimal()` to create a new s2n_config object without loading system
+ * certificates by default.
+ *
+ * The returned config should be freed with `s2n_config_free()` after it's no longer in use by any
+ * connection.
+ *
+ * @returns A new configuration object suitable for configuring connections and associating certs
+ * and keys.
  */
 S2N_API extern struct s2n_config *s2n_config_new(void);
 
+/**
+ * Creates a new s2n_config object with minimal default options.
+ *
+ * This API is similar to `s2n_config_new()`, except that the returned config is created without
+ * loading default system certificates into the trust store. To add system certificates to this
+ * config, call `s2n_config_load_system_certs()`.
+ *
+ * The returned config should be freed with `s2n_config_free()` after it's no longer in use by any
+ * connection.
+ *
+ * @returns A new configuration object suitable for configuring connections and associating certs
+ * and keys.
+ */
+S2N_API extern struct s2n_config *s2n_config_new_minimal(void);
+
```

Comment:
After reading all your documentation, I'm wondering if this was the right name to choose. Maybe we want a name that is very clear about what "minimal" actually means? Like "s2n_config_new_with_empty_trust_store" but maybe more succinct. Or maybe we just need to stress that "s2n_config_new_minimal" is more about performance? Or maybe not stress the impact on the trust store so much, since the existing pattern for that is s2n_config_new + s2n_config_wipe_trust_store?

Reply:
I think the best case would be if there's a name that can basically replace s2n_config_new, but I'm not sure one exists... Maybe s2n_config_alloc or s2n_config_new_empty? Without a better name, I kept s2n_config_new_minimal and just recommended it for performance reasons, and kept s2n_config_new for the normal case.

Reply:
If we add a future optimization to s2n_config_new_minimal, it will probably have to be something new that's added to s2n_config_new and not s2n_config_new_minimal, and isn't be a breaking change (otherwise it will need a new API). So, it might be okay to mention the trust store here specifically, since we can always update the s2n_config_new_minimal documentation with the future change?

I do mention performance in the API documentation and in the usage guide:
https://github.com/aws/s2n-tls/blob/a6e8384f29856cfa7e435a425c7297be26962a94/api/s2n.h#L260
https://github.com/aws/s2n-tls/blob/a6e8384f29856cfa7e435a425c7297be26962a94/docs/USAGE-GUIDE.md?plain=1#L402-L403

Let me know if you have a name that indicates the performance that's better than s2n_config_new_minimal!

Reply:
`s2n_config_new_minimal` seems reasonable to me. I think we should avoid the use of any specific feature (trust_store) so that its a bit more future-proofed.

Reply:
Do you mean removing the note about the trust store in the s2n_config_new_minimal documentation? I'm not sure this should be removed, since it's related to our recommendation on when to call `s2n_config_wipe_trust_store`. The user has to know that the config isn't initialized with default certs, so they know it's safe to not call  `s2n_config_wipe_trust_store`.

I'm also not sure if this hurts the API in the future. We wouldn't be able to start initializing the trust store with this API anyway, since that would be a breaking change.

Reply:
I meant only avoiding it in the name.